### DISCUSSION TITLE
Cloud Foundry V3 API - Authentication

### DISCUSSION THREAD

**Discussion by mash-sap** [cloudfoundry/cli#2379] [Q&A]
Hi everyone,

we want to programatically access cf apis:
https://v3-apidocs.cloudfoundry.org/version/3.132.0/index.html#authentication

Our question is related to authentication. We are using below request for getting the oauth token. But we could not find any documentation on this client_id ("cf"). Is there any documentation available? Are there other options to get the oauth token without installing the CLI?

Thanks in advance &
Best regards,
Manuel

We get the token via HTTP request:
POST https://login.cf.<host>/oauth/token
grant_type=password
**client_id = cf**
client_secret=
username=mySpaceDev@fu.bar
passowrd=...

**Comment by moleske** [comment]
The oauth token comes from UAA as mentioned in the authentication docs you linked.  You should check out the linked UAA docs to understand how to get an oauth token https://docs.cloudfoundry.org/api/uaa/version/76.6.0/index.html#token

**Reply by mash-sap** [reply]
Thanks for your reply. We were already able to get a token. We are using Password Grant with above mentioned client_id/secret.
https://docs.cloudfoundry.org/api/uaa/version/76.6.0/index.html#password-grant

The question is regarding the client_id "cf". I could not find any documentation on it. But it seems to be used by clients like:
https://github.com/cloudfoundry/cli (this project)
https://github.com/cloudfoundry/cf-java-client

So would it be fine to rely on this client_id? Is there any documentation that we can refer to?

**Reply by mash-sap** [reply]
Can someone clarify?

**Comment by stephanme** [comment]
I discussed the topic in CAPI Open Office Hour and in the ARD Working Group Meeting. There was no objection to document the UAA client `cf` so that it can be used to obtain an OAuth2 token for programmatic CF API access. Same procedure as done by CF CLI. Especially in PaaS foundations, users don't have the option to create/obtain own UAA clients.

I opened https://github.com/cloudfoundry/docs-cloudfoundry-concepts/pull/182 

Not sure if this is the perfect place in the documentation but it seems to me a good starting point. Later, it can be additionally documented in e.g. https://v3-apidocs.cloudfoundry.org/index.html#authentication
